### DISCUSSION TITLE
Grafana integration

### DISCUSSION THREAD

**Discussion by Alan-Logic** [pelican-dev/panel#44] [Ideas]
Grafana integration would make pelican panel better for Developers and  game server owners.  What Grafana used for?

I love how you can see cpu ram useage and all but I think  something like Grafana would be a nice optional future for  debugging things like  
Network problems,  game plugin problems, cpu spikes  in-depth charts that display your servers performance would help a lot of community game servers.

This would be great for games like

Rust 
ARK
CSGO
G mod 
and more!

**Comment by wdhdev** [comment]
This is achievable by using InfluxDB. You can create a Telegraf config and install it on the host machine (or in this case your panel). And then you can link your InfluxDB instance to Grafana and use a dashboard like https://grafana.com/grafana/dashboards/928-telegraf-system-dashboard/ to view the information.

**Comment by CodeExplorerX** [comment]
We already have something similar for Pterodactyl. I think we could use it as a basis. And there's no need to install InfluxDB and Telegraf here, which burden the virtual server, especially InfluxDB. Just Prometheus and this exporter would be enough.
https://github.com/LOENS2/pterodactyl_exporter
https://grafana.com/grafana/dashboards/16575-pterodactyl-server/

**Reply by Exos9** [reply]
> We already have something similar for Pterodactyl. I think we could use it as a basis. And there's no need to install InfluxDB and Telegraf here, which burden the virtual server, especially InfluxDB. Just Prometheus and this exporter would be enough. https://github.com/LOENS2/pterodactyl_exporter https://grafana.com/grafana/dashboards/16575-pterodactyl-server/

After doing a bit of testing, it seems that this works perfectly fine on Pelican, even though there is no official support yet. It would be very cool to have a Prometheus exporter baked into the panel though

**Comment by lancepioch** [comment]
This would serve as a fantastic future plugin!

**Comment by LOENS2** [comment]
I would be willing to add Pelican support to the pterodactyl_exporter or even create an entirely new script for that. However at the moment, I think that the project is still to immature to support it.
